src/client/Client.cc

```diff
@@ -2718,34 +2722,54 @@ void Client::handle_mds_map(const MConstRef<MMDSMap>& m)
        i != cancel_ops.end(); ++i) {
     command_table.erase(*i);
   }
+}
+
+void Client::handle_mds_map(const MConstRef<MMDSMap>& m)
+{
+  std::unique_lock cl{client_lock, std::adopt_lock};
```

Comment:
This is just weird. Let's change the `handle_*` methods to take the client_lock themselves only if they need it.